packages/iles/src/node/config.ts

```diff
@@ -182,9 +182,21 @@ async function createIlesModule (pkgName: string, ...options: any[]): Promise<Il
 function inferJSX (config: UserConfig) {
   const plugins = config.vite?.plugins?.flat() || []
   for (const plugin of plugins) {
-    const { name = '' } = plugin || {}
-    if (name.includes('preact')) return 'preact'
-    if (name.includes('solid')) return 'solid'
+    if (!plugin)
+      continue
+
+    if (Array.isArray(plugin)) {
+      for (const plugin2 of plugin) {
+        const { name = '' } = plugin2 && !Array.isArray(plugin2) ? plugin2 : {}
+        if (name.includes('preact')) return 'preact'
+        if (name.includes('solid')) return 'solid'
+      }
+    }
+    else {
+      const { name = '' } = plugin || {}
+      if (name.includes('preact')) return 'preact'
+      if (name.includes('solid')) return 'solid'
+    }
```

Comment:
Is `flat` not doing the trick? Perhaps we can pass `Infinity` to flatten all levels?